sway/desktop/render.c

```diff
@@ -158,6 +159,14 @@ static void render_surface_iterator(struct sway_output *output, struct sway_view
 	wlr_matrix_project_box(matrix, &dst_box, transform, rotation,
 		wlr_output->transform_matrix);
 
+	struct sway_container *container = data->container;
+	if (container != NULL) {
+		dst_box.width = fmin(dst_box.width,
+				wlr_output->scale * (container->current.content_width - surface->sx));
```

Comment:
Is there a reason why we can't do this before the `scale_box` call? Is it because the projection needs the full non-clipped box?

Reply:
Clipping the box before the projection leads to a transformation matrix that squeezes the surface to fit rather than clipping it on overflow...

Which is a shame, as the code was a lot prettier when I could clip before scale_box. :(

Reply:
Maybe the scaling in the `fmin` calls needs some adjustment by the way, in order to be scaled equivalently to how `dst_box_width` is scaled?

Reply:
Can you elaborate on what your mean by this?

Reply:
> Ah, you mean the rounding logic in scale_box?
Yep!